tests/lib/collectors/libcloudforensics_e2e.py

```diff
@@ -0,0 +1,246 @@
+    gcloud_collector = gcloud.GoogleCloudCollector(self.test_state)
+    self.assertIsNotNone(gcloud_collector)
+
+    # This should make a copy of the boot disk only
+    gcloud_collector.SetUp(
+        self.project_id,
+        self.project_id,
+        self.incident_id,
+        self.zone,
+        42.0,
+        16,
+        remote_instance_name=self.instance_to_analyse,
+    )
+
+    self.analysis_vm = gcloud_collector.analysis_vm
+
+    # Check that the collector is correctly initialized
+    self.assertEqual(self.test_state.errors, [])
+    self.assertEqual(gcloud_collector.disk_names, [])
+    self.assertEqual(gcloud_collector.analysis_project.project_id,
+                     self.project_id)
+    self.assertEqual(gcloud_collector.remote_project.project_id,
+                     self.project_id)
+    self.assertEqual(gcloud_collector.remote_instance_name,
+                     self.instance_to_analyse)
+    self.assertEqual(gcloud_collector.all_disks, False)
+
+    # Attach the disk copy to the analysis VM
+    gcloud_collector.Process()
+
+    self.assertEqual(
+        self.test_state.output[0][0], 'gcp-forensics-vm-fake-incident-id')
+    self.assertIsInstance(self.test_state.output[0][1], gcp.GoogleComputeDisk)
+    self.assertTrue(self.test_state.output[0][1].name.startswith(
+        'incidentfake-incident-id'))
+    self.assertTrue(self.test_state.output[0][1].name.endswith('-copy'))
+    self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+    self.assertEqual(self.analysis_vm.name, 'gcp-forensics-vm-fake-incident-id')
+
+    disks = self.analysis_vm.list_disks()
+    self.assertEqual(
+        disks,
+        ['gcp-forensics-vm-fake-incident-id',
+         self.test_state.output[0][1].name])
+
+    if self.disk_to_forensic is None:
+      self.__clean()
+      return
+
+    # This should make a copy of the disk specified in 'disk-names'
+    gcloud_collector.SetUp(
+        self.project_id,
+        self.project_id,
+        self.incident_id,
+        self.zone,
+        42.0,
+        16,
+        remote_instance_name=self.instance_to_analyse,
+        disk_names=self.disk_to_forensic
+    )
+
+    self.analysis_vm = gcloud_collector.analysis_vm
+
+    # Attach the disk copy to the analysis VM
+    gcloud_collector.Process()
+
+    self.assertEqual(
+        self.test_state.output[1][0], 'gcp-forensics-vm-fake-incident-id')
+    self.assertIsInstance(self.test_state.output[1][1], gcp.GoogleComputeDisk)
+    self.assertTrue(self.test_state.output[1][1].name.startswith(
+        'incidentfake-incident-id'))
+    self.assertTrue(self.test_state.output[1][1].name.endswith('-copy'))
+
+    disks = self.analysis_vm.list_disks()
+    self.assertEqual(disks,
+                     ['gcp-forensics-vm-fake-incident-id',
+                      self.test_state.output[0][1].name,
+                      self.test_state.output[1][1].name])
+
+    self.__clean()
+
+  def __clean(self):
```

Comment:
Consider the case where we'll have more than one end-to-end test (e.g with different parameters passed in SetUp()), you may want to call this method `tearDown()` ([link](https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown)) as it will be called after each test method invocation (the same happens with `setUp()`)

Reply:
Addressed, I separated the test in two for better readability.